src/UnigramWordPredictor.java

```diff
@@ -50,8 +51,22 @@ public UnigramWordPredictor(Tokenizer tokenizer) {
    */
   public void train(Scanner scanner) {
     List<String> trainingWords = tokenizer.tokenize(scanner);
+    Map<String, List<String>> wordMap = new HashMap<>();
 
-    // TODO: Convert the trainingWords into neighborMap here
+
+    for(int i = 0; i < trainingWords.size()-1;i++){
+      String wordKey = trainingWords.get(i);
+      String nextWord = trainingWords.get(i+1);
+
+      if(!wordMap.containsKey(wordKey)){
+        List<String> wordList = new ArrayList<>();
+        wordList.add(nextWord);
+        wordMap.put(wordKey, wordList);
+      }else{
+        wordMap.get(wordKey).add(nextWord);
+      }
+    }
```

Comment:
Nice! As an alternative, can you think about how you could write this using `putIfAbsent` or `computeIfAbsent` instead?